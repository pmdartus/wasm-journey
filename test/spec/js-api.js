const path = require('path');
const { run } = require('./runner');

run({
    extension: '.any.js',
    base: path.resolve(__dirname, 'wasm/test/js-api'),
    expected: path.resolve(__dirname, './js-api.expected.json'),

    mapping: {
        // There is an explicit mapping because the test structure in the WASM directory differs
        // from the test structure in the official WPT repo.
        '/wasm/jsapi': path.resolve(__dirname, 'wasm/test/js-api'),
    },

    ignore: [
        // Takes too long to run. Let's revisit this once the limits are in place.
        'limits.any.js',
    ],
});
