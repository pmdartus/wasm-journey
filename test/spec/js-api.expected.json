{
    "constructor/instantiate-bad-imports.any.js": [
        "WebAssembly.instantiate(module): Non-object imports argument: null",
        "WebAssembly.instantiate(module): Non-object imports argument: true",
        "WebAssembly.instantiate(module): Non-object imports argument: \"\"",
        "WebAssembly.instantiate(module): Non-object imports argument: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Non-object imports argument: 1",
        "WebAssembly.instantiate(module): Non-object imports argument: 0.1",
        "WebAssembly.instantiate(module): Non-object imports argument: NaN",
        "WebAssembly.instantiate(module): Non-object module: undefined",
        "WebAssembly.instantiate(module): Non-object module: null",
        "WebAssembly.instantiate(module): Non-object module: true",
        "WebAssembly.instantiate(module): Non-object module: \"\"",
        "WebAssembly.instantiate(module): Non-object module: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Non-object module: 1",
        "WebAssembly.instantiate(module): Non-object module: 0.1",
        "WebAssembly.instantiate(module): Non-object module: NaN",
        "WebAssembly.instantiate(module): Missing imports argument",
        "WebAssembly.instantiate(module): Imports argument with missing property: undefined",
        "WebAssembly.instantiate(module): Imports argument with missing property: empty object",
        "WebAssembly.instantiate(module): Imports argument with missing property: wrong property",
        "WebAssembly.instantiate(module): Importing an i64 global",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: null",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: true",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(module): Importing a function with an incorrectly-typed value: object \"[object Object]\"",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: null",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: true",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: WebAssembly.Global",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: WebAssembly.Global.prototype",
        "WebAssembly.instantiate(module): Importing a global with an incorrectly-typed value: Object.create(WebAssembly.Global.prototype)",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: null",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: true",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: WebAssembly.Memory",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: WebAssembly.Memory.prototype",
        "WebAssembly.instantiate(module): Importing memory with an incorrectly-typed value: Object.create(WebAssembly.Memory.prototype)",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: null",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: true",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: WebAssembly.Table",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: WebAssembly.Table.prototype",
        "WebAssembly.instantiate(module): Importing table with an incorrectly-typed value: Object.create(WebAssembly.Table.prototype)",
        "WebAssembly.instantiate(buffer): Non-object imports argument: null",
        "WebAssembly.instantiate(buffer): Non-object imports argument: true",
        "WebAssembly.instantiate(buffer): Non-object imports argument: \"\"",
        "WebAssembly.instantiate(buffer): Non-object imports argument: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Non-object imports argument: 1",
        "WebAssembly.instantiate(buffer): Non-object imports argument: 0.1",
        "WebAssembly.instantiate(buffer): Non-object imports argument: NaN",
        "WebAssembly.instantiate(buffer): Non-object module: undefined",
        "WebAssembly.instantiate(buffer): Non-object module: null",
        "WebAssembly.instantiate(buffer): Non-object module: true",
        "WebAssembly.instantiate(buffer): Non-object module: \"\"",
        "WebAssembly.instantiate(buffer): Non-object module: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Non-object module: 1",
        "WebAssembly.instantiate(buffer): Non-object module: 0.1",
        "WebAssembly.instantiate(buffer): Non-object module: NaN",
        "WebAssembly.instantiate(buffer): Missing imports argument",
        "WebAssembly.instantiate(buffer): Imports argument with missing property: undefined",
        "WebAssembly.instantiate(buffer): Imports argument with missing property: empty object",
        "WebAssembly.instantiate(buffer): Imports argument with missing property: wrong property",
        "WebAssembly.instantiate(buffer): Importing an i64 global",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: null",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: true",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(buffer): Importing a function with an incorrectly-typed value: object \"[object Object]\"",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: null",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: true",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: WebAssembly.Global",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: WebAssembly.Global.prototype",
        "WebAssembly.instantiate(buffer): Importing a global with an incorrectly-typed value: Object.create(WebAssembly.Global.prototype)",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: null",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: true",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: WebAssembly.Memory",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: WebAssembly.Memory.prototype",
        "WebAssembly.instantiate(buffer): Importing memory with an incorrectly-typed value: Object.create(WebAssembly.Memory.prototype)",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: undefined",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: null",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: true",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: \"\"",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: symbol \"Symbol()\"",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: 1",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: 0.1",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: NaN",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: plain object",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: WebAssembly.Table",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: WebAssembly.Table.prototype",
        "WebAssembly.instantiate(buffer): Importing table with an incorrectly-typed value: Object.create(WebAssembly.Table.prototype)"
    ],
    "constructor/instantiate.any.js": [
        "getter order for imports object: BufferSource argument",
        "getter order for imports object: Module argument",
        "imports: BufferSource argument",
        "imports: Module argument",
        "No imports: BufferSource argument",
        "No imports: Module argument",
        "exports and imports: BufferSource argument",
        "exports and imports: Module argument",
        "Synchronous options handling: Buffer argument",
        "Synchronous options handling: Module argument"
    ],
    "global/constructor.any.js": [
        "length",
        "No arguments",
        "Order of evaluation",
        "Invalid descriptor argument",
        "Invalid type argument",
        "i64 with default",
        "Default value for type i32",
        "Explicit value undefined for type i32",
        "Explicit value null for type i32",
        "Explicit value true for type i32",
        "Explicit value false for type i32",
        "Explicit value 2 for type i32",
        "Explicit value \"3\" for type i32",
        "Explicit value object with toString for type i32",
        "Explicit value object with valueOf for type i32",
        "Default value for type f32",
        "Explicit value undefined for type f32",
        "Explicit value null for type f32",
        "Explicit value true for type f32",
        "Explicit value false for type f32",
        "Explicit value 2 for type f32",
        "Explicit value \"3\" for type f32",
        "Explicit value object with toString for type f32",
        "Explicit value object with valueOf for type f32",
        "Default value for type f64",
        "Explicit value undefined for type f64",
        "Explicit value null for type f64",
        "Explicit value true for type f64",
        "Explicit value false for type f64",
        "Explicit value 2 for type f64",
        "Explicit value \"3\" for type f64",
        "Explicit value object with toString for type f64",
        "Explicit value object with valueOf for type f64",
        "Stray argument"
    ],
    "global/toString.any.js": [
        "Object.prototype.toString on an Global"
    ],
    "global/value-get-set.any.js": [
        "Branding",
        "Immutable i32 (missing)",
        "Immutable i32 (undefined)",
        "Immutable i32 (null)",
        "Immutable i32 (false)",
        "Immutable i32 (empty string)",
        "Immutable i32 (zero)",
        "Mutable i32 (true)",
        "Mutable i32 (one)",
        "Mutable i32 (string)",
        "Mutable i32 (true on prototype)",
        "Immutable f32 (missing)",
        "Immutable f32 (undefined)",
        "Immutable f32 (null)",
        "Immutable f32 (false)",
        "Immutable f32 (empty string)",
        "Immutable f32 (zero)",
        "Mutable f32 (true)",
        "Mutable f32 (one)",
        "Mutable f32 (string)",
        "Mutable f32 (true on prototype)",
        "Immutable f64 (missing)",
        "Immutable f64 (undefined)",
        "Immutable f64 (null)",
        "Immutable f64 (false)",
        "Immutable f64 (empty string)",
        "Immutable f64 (zero)",
        "Mutable f64 (true)",
        "Mutable f64 (one)",
        "Mutable f64 (string)",
        "Mutable f64 (true on prototype)",
        "i64 with default",
        "Calling setter without argument",
        "Stray argument"
    ],
    "global/valueOf.any.js": [
        "Branding",
        "Stray argument"
    ],
    "instance/constructor-bad-imports.any.js": [
        "new WebAssembly.Instance(module): Non-object imports argument: null",
        "new WebAssembly.Instance(module): Non-object imports argument: true",
        "new WebAssembly.Instance(module): Non-object imports argument: \"\"",
        "new WebAssembly.Instance(module): Non-object imports argument: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Non-object imports argument: 1",
        "new WebAssembly.Instance(module): Non-object imports argument: 0.1",
        "new WebAssembly.Instance(module): Non-object imports argument: NaN",
        "new WebAssembly.Instance(module): Non-object module: undefined",
        "new WebAssembly.Instance(module): Non-object module: null",
        "new WebAssembly.Instance(module): Non-object module: true",
        "new WebAssembly.Instance(module): Non-object module: \"\"",
        "new WebAssembly.Instance(module): Non-object module: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Non-object module: 1",
        "new WebAssembly.Instance(module): Non-object module: 0.1",
        "new WebAssembly.Instance(module): Non-object module: NaN",
        "new WebAssembly.Instance(module): Imports argument with missing property: empty object",
        "new WebAssembly.Instance(module): Imports argument with missing property: wrong property",
        "new WebAssembly.Instance(module): Importing an i64 global",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: undefined",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: null",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: true",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: \"\"",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: 1",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: 0.1",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: NaN",
        "new WebAssembly.Instance(module): Importing a function with an incorrectly-typed value: object \"[object Object]\"",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: undefined",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: null",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: true",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: \"\"",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: plain object",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: WebAssembly.Global",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: WebAssembly.Global.prototype",
        "new WebAssembly.Instance(module): Importing a global with an incorrectly-typed value: Object.create(WebAssembly.Global.prototype)",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: undefined",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: null",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: true",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: \"\"",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: 1",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: 0.1",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: NaN",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: plain object",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: WebAssembly.Memory",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: WebAssembly.Memory.prototype",
        "new WebAssembly.Instance(module): Importing memory with an incorrectly-typed value: Object.create(WebAssembly.Memory.prototype)",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: undefined",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: null",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: true",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: \"\"",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: symbol \"Symbol()\"",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: 1",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: 0.1",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: NaN",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: plain object",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: WebAssembly.Table",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: WebAssembly.Table.prototype",
        "new WebAssembly.Instance(module): Importing table with an incorrectly-typed value: Object.create(WebAssembly.Table.prototype)"
    ],
    "instance/constructor.any.js": [
        "length",
        "getter order for imports object",
        "imports",
        "No imports",
        "exports and imports"
    ],
    "instance/exports.any.js": [
        "Branding"
    ],
    "memory/buffer.any.js": [
        "Branding"
    ],
    "memory/constructor.any.js": [
        "Order of evaluation for descriptor",
        "Non-zero initial"
    ],
    "memory/grow.any.js": [
        "Missing arguments",
        "Branding",
        "Zero initial",
        "Zero initial with valueOf",
        "Non-zero initial",
        "Zero initial with respected maximum",
        "Zero initial with respected maximum grown twice",
        "Zero initial growing too much",
        "Out-of-range argument: undefined",
        "Out-of-range argument: NaN",
        "Out-of-range argument: Infinity",
        "Out-of-range argument: -Infinity",
        "Out-of-range argument: -1",
        "Out-of-range argument: 4294967296",
        "Out-of-range argument: 68719476736",
        "Out-of-range argument: \"0x100000000\"",
        "Out-of-range argument: object \"[object Object]\"",
        "Stray argument"
    ],
    "module/customSections.any.js": [
        "Custom sections with surrogate pairs",
        "Custom sections with U+FFFD"
    ],
    "module/exports.any.js": [
        "exports"
    ],
    "module/imports.any.js": [
        "Missing arguments",
        "Non-Module arguments",
        "imports"
    ],
    "table/constructor.any.js": [
        "No arguments",
        "Empty descriptor",
        "Invalid descriptor argument",
        "Undefined initial value in descriptor",
        "Undefined element value in descriptor",
        "Out-of-range initial value in descriptor: NaN",
        "Out-of-range maximum value in descriptor: NaN",
        "Out-of-range initial value in descriptor: Infinity",
        "Out-of-range maximum value in descriptor: Infinity",
        "Out-of-range initial value in descriptor: -Infinity",
        "Out-of-range maximum value in descriptor: -Infinity",
        "Out-of-range initial value in descriptor: -1",
        "Out-of-range maximum value in descriptor: -1",
        "Out-of-range initial value in descriptor: 4294967296",
        "Out-of-range maximum value in descriptor: 4294967296",
        "Out-of-range initial value in descriptor: 68719476736",
        "Out-of-range maximum value in descriptor: 68719476736",
        "Initial value exceeds maximum",
        "Basic (non-zero)",
        "Type conversion for descriptor.element",
        "Order of evaluation for descriptor"
    ],
    "table/get-set.any.js": [
        "Missing arguments: get",
        "Branding: get",
        "Missing arguments: set",
        "Branding: set",
        "Basic",
        "Growing",
        "Setting out-of-bounds",
        "Setting non-function",
        "Setting non-wasm function",
        "Setting non-wasm arrow function",
        "Getting out-of-range argument: undefined",
        "Setting out-of-range argument: undefined",
        "Getting out-of-range argument: NaN",
        "Setting out-of-range argument: NaN",
        "Getting out-of-range argument: Infinity",
        "Setting out-of-range argument: Infinity",
        "Getting out-of-range argument: -Infinity",
        "Setting out-of-range argument: -Infinity",
        "Getting out-of-range argument: -1",
        "Setting out-of-range argument: -1",
        "Getting out-of-range argument: 4294967296",
        "Setting out-of-range argument: 4294967296",
        "Getting out-of-range argument: 68719476736",
        "Setting out-of-range argument: 68719476736",
        "Getting out-of-range argument: \"0x100000000\"",
        "Setting out-of-range argument: \"0x100000000\"",
        "Getting out-of-range argument: object \"[object Object]\"",
        "Setting out-of-range argument: object \"[object Object]\"",
        "Order of argument conversion",
        "Stray argument"
    ],
    "table/grow.any.js": [
        "Missing arguments",
        "Branding",
        "Basic",
        "Reached maximum",
        "Exceeded maximum",
        "Out-of-range argument: undefined",
        "Out-of-range argument: NaN",
        "Out-of-range argument: Infinity",
        "Out-of-range argument: -Infinity",
        "Out-of-range argument: -1",
        "Out-of-range argument: 4294967296",
        "Out-of-range argument: 68719476736",
        "Out-of-range argument: \"0x100000000\"",
        "Out-of-range argument: object \"[object Object]\"",
        "Stray argument"
    ],
    "table/length.any.js": [
        "Branding",
        "Stray argument",
        "Setting (sloppy mode)",
        "Setting (strict mode)"
    ]
}